bindings/matrix-sdk-crypto-ffi/src/verification.rs

```diff
@@ -425,6 +497,45 @@ pub struct ConfirmVerificationResult {
     pub signature_request: Option<SignatureUploadRequest>,
 }
 
+/// Listener that will be passed over the FFI to report changes to a
+/// verification request.
+pub trait VerificationRequestListener: Send {
+    /// The callback that should be called on the Rust side
+    ///
+    /// # Arguments
+    ///
+    /// * `state` - The current state of the verification request.
+    fn on_change(&self, state: VerificationRequestState);
+}
+
+/// An Enum describing the state the QrCode verification is in.
+pub enum VerificationRequestState {
+    /// The verification request was sent
+    Requested,
```

Comment:
Don't we need to know the methods that are supported here? For SAS it was unimportant since we could make the emojis optional and if they aren't there the clients don't support them.

How do you decide in Element if you're going to show a QR code or not?

Reply:
Oh indeed, the methods do change during the lifetime of the request. Added now, I think we only need to know for `Ready` state